lib/Polaris.Class.ps1

```diff
@@ -57,7 +58,13 @@ class Polaris {
                 $Polaris.Log("Parsed Route: $Route")
                 $Polaris.Log("Request Method: $($RawRequest.HttpMethod)")
                 $Routes = $Polaris.ScriptblockRoutes
-                $MatchingRoute = $Routes.keys.Where( { $Route -match $_ })[0]
+
+                #
+                # Searching for the first route that matches by the most specific route paths first.
+                #
+                $MatchingRoute = $Routes.keys | Sort-Object -Property Length -Descending | Where-Object { $Route -match [Polaris]::ConvertPathToRegex($_) } | Select-Object -First 1
+                $Request.Parameters = ([PSCustomObject]$Matches)
```

Comment:
Where does `$Matches` get defined?

Reply:
$Matches is an automatic variable that gets defined every time you run a -match comparison. 

See:
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-6#matches
https://mcpmag.com/articles/2015/09/30/regex-groups-with-powershell.aspx